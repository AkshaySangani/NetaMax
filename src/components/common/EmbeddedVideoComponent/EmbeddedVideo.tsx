import { ReactElement } from "react";

import YouTube from "react-youtube";

import { screenSizes } from "styled/screen";

import { useMediaQuery } from "@chakra-ui/react";

import { IEmbeddedVideoProps } from "./IEmbeddedVideoProps";

/**
 *  The EmbeddedVideo component.
 * @param {IEmbeddedVideoProps} props The props of the component.
 * @returns {ReactElement} The EmbeddedVideo component.
 */
export const EmbeddedVideo = (props: IEmbeddedVideoProps): ReactElement => {
  const { videoUrlId, onPlayVideoClick } = props;
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.sm}px)`);

  const opts = {
    height: isDesktop ? "390" : "190",
    width: isDesktop ? "640" : "310",
    playerVars: {},
  };

  return (
    <YouTube
      videoId={videoUrlId}
      title="Youtube video"
      opts={opts}
      onPlay={onPlayVideoClick}
      containerStyle={{ height: "100%", top: "100%" }}
    />
  );
};
