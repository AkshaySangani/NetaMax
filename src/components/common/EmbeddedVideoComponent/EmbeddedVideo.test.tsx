import { render } from "@testing-library/react";
import { EmbeddedVideo } from "./EmbeddedVideo";

jest.mock("next/router", () => require("next-router-mock"));

describe("Render EmbeddedVideo component", () => {
  it("EmbeddedVideo should be render correctly", () => {
    const { container } = render(<EmbeddedVideo videoUrlId="D-OJLyECbfo" />);
    expect(container).toMatchSnapshot();
  });
});
