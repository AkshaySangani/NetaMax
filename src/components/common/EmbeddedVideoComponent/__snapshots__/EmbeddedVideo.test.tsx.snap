// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render EmbeddedVideo component EmbeddedVideo should be render correctly 1`] = `
<div>
  <div
    class=""
    style="height: 100%; top: 100%;"
  >
    <div
      class=""
      id=""
    />
  </div>
</div>
`;
